dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -65,16 +67,25 @@ async def snapshot(self, req):
         return dashboard_utils.rest_response(
             success=True, message="hello", snapshot=snapshot)
 
+    def _get_job_status(self, metadata: Dict[str, str]) -> Optional[str]:
+        status = None
+        job_submission_id = metadata.get(JOB_ID_METADATA_KEY)
+        if job_submission_id is not None:
+            status = str(self._job_status_client.get_status(job_submission_id))
```

Comment:
is it possible that this isn't set / returns None?

Reply:
If there's a job submission ID, this must be set

Reply:
added comment